rust/docs/parsing.md

```
@@ -0,0 +1,170 @@
+and root rules named `header` are treated specially under the following conditions.
+
+A `header`, if preceded by a newline (or at start of input), is a log event separator.
+Otherwise, it is treated as an ordinary rule.
+
+If no separating `header` has been encountered (yet), log events are separated on newlines.
+
+Newlines are part of the line preceding it;
+in other words, log events are always terminated by newlines
+(but a newline doesn't necessarily terminate a log event).
+
+### TODO
```

markdownlint-cli2 (0.22.1)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)